lib/growl.js

```diff
@@ -0,0 +1,151 @@
+'use strict';
+/**
+ * @module Growl
+ */
+
+var fs = require('fs');
+var os = require('os');
+var path = require('path');
+
+/**
+ * @summary
+ * Checks if Growl support seems likely.
+ *
+ * @description
+ * Glosses over the distinction between an unsupported platform
+ * and one that lacks prerequisite software installations.
+ * Autofails when run in browser.
+ *
+ * @public
+ * @see {@link https://github.com/tj/node-growl/blob/master/README.md|Prerequisite Installs}
+ * @see {@link Mocha#growl}
+ * @see {@link Mocha#isGrowlCapable}
+ * @return {boolean} whether Growl support can be expected
+ */
+exports.isCapable = function() {
+  return !process.browser && which(getSupportBinaries()) !== '';
+};
+
+/**
+ * Implements desktop notifications as a pseudo-reporter.
+ *
+ * @public
+ * @see {@link Mocha#_growl}
+ * @param {Runner} runner - Runner instance.
+ */
+exports.notify = function(runner) {
+  var sendNotification = function() {
+    var growl = require('growl');
+    var stats = runner.stats;
+    var msg;
+    var options;
+
+    if (stats.failures) {
+      msg = stats.failures + ' of ' + runner.total + ' tests failed';
+      options = {
+        name: 'mocha',
+        title: 'Failed',
+        image: image('error')
+      };
+    } else {
+      msg = stats.passes + ' tests passed in ' + stats.duration + 'ms';
+      options = {
+        name: 'mocha',
+        title: 'Passed',
+        image: image('ok')
+      };
+    }
+    growl(msg, options, onCompletion);
+  };
+
+  /**
+   * Callback for result of attempted Growl notification.
+   *
+   * @callback Growl~growlCB
+   * @param {*} err - Error object, or <code>null</code> if successful.
+   * @param {string} stdout - <code>stdout</code> from notification delivery
+   *     process.
+   * @param {string} stderr - <code>stderr</code> from notification delivery
+   *     process. It will include timestamp and executed command with arguments.
+   */
+
+  function onCompletion(err, stdout, stderr) {
+    if (err) {
+      if (err.code === 'ENOENT') {
+        console.error(
+          'Growl notification support not installed... listener removed!'
+        );
+        runner.removeListener('end', sendNotification);
```

Comment:
So we had this chunk of code from #3311, but I now wonder about its utility.
L74 should now only occur if the user uninstalls the software _during_ the run (unlikely). Even so, Growl is only invoked _after_ all tests are done. Does it buy anything to bother removing a listener that has completed a one-time event?